base/tuple.jl

```diff
@@ -14,6 +14,11 @@ true
 """
 NTuple
 
+# convenience function for extracting N from a Tuple (if defined)
+# else return `nothing` for anything else given (such as Vararg or other non-sized Union)
+_counttuple(::Type{<:NTuple{N,Any}}) where {N} = N
+_counttuple(::Type) = nothing
```

Comment:
Per the discussion here:
https://github.com/JuliaLang/julia/pull/32495#issuecomment-508276553

Can we name this something not private and export it? This is something we use all the time in our code at RelationalAI, where we do a lot of stuff with tuples. It also seems like something that other code will use a lot, like DataFrames, CSV, etc.

In that other comment I suggested `arity(::Type{<:Tuple})` as one suggestion. I still like that best, but open to others!
- `arity()`
- `tupletypewidth()` / `tuple_type_width()`
- `tupletypecount()` / `tuple_type_count()`
- `tuple_type_width()`

Or maybe we can break out a separate issue to track this, so that it doesn't hold up this PR?

Reply:
Yeah, I made this slightly internal just so it wouldn't delay this PR. Makes sense to me to track this as a separate issue to consider exporting it!

Reply:
Thanks. Opened https://github.com/JuliaLang/julia/issues/36820.

Resolving.